recursive_art.py

```diff
@@ -20,11 +20,26 @@ def build_random_function(min_depth, max_depth):
         (See the assignment writ-eup for details on the representation of
         these functions)
     """
-    # TODO: implement this
-    pass
+    functions = ['prod', 'avg', 'acos_pi', 'asin_pi', 'cos_pi', 'sin_pi', 'cubed'] # possible operations
+
+    if movie:
+        variables = [['x'],['y'],['t']]  # three variables when making a movie
+    else:
+        variables = [['x'],['y']]
+
+    if max_depth == 0:
+        return variables[random.randint(0,len(variables)-1)]  # if maxed depth is reached, stop recursion
+
+    rand_func = functions[random.randint(0,len(functions)-1)]  # pick a random function
+
+    if rand_func in functions[0:4]: # for the two-variable functions
+        return [rand_func, build_random_function(min_depth-1,max_depth-1,movie), build_random_function(min_depth-1,max_depth-1,movie)]
+    else:  # for one variable functions
+        return [rand_func, build_random_function(min_depth-1,max_depth-1,movie)]
 
 
-def evaluate_random_function(f, x, y):
+
+def evaluate_random_function(f, x, y, t):
```

Comment:
I recommend that you add more unit tests of your own to test functions like this in future projects.